controllers/vcdcluster_controller.go

```diff
@@ -395,10 +395,12 @@ func getMapOfEdgeGateways(ctx context.Context, vcdClient *vcdsdk.Client, orgName
 func (r *VCDClusterReconciler) getOvdcList(ctx context.Context, vcdOrg *govcd.Org, vcdCluster *infrav1beta3.VCDCluster,
 	vdc *govcd.Vdc) ([]rdeType.Ovdc, error) {
 
-	if vdc == nil && vcdCluster.Spec.MultiZoneSpec.Zones == nil {
+	var err error
+	if vdc == nil && vcdCluster.Spec.Ovdc == "" && vcdCluster.Spec.MultiZoneSpec.Zones == nil {
```

Comment:
vcdCluster.Spec.MultiZoneSpec.Zones is of type []Zone, not *[]Zone. So, do we not want to check for nil, and instead want this check to be for len(vcdCluster.Spec.MultiZoneSpec.Zones) == 0

Reply:
Thanks @abaruni . I'll scan the codebase and make these change.